vars/fhBuildNode.groovy

```diff
@@ -2,9 +2,10 @@
 
 def call(Map parameters = [:], body) {
 
-    def label = parameters.get('label', 'nodejs-ubuntu')
+    def labels = [params0.get('label')].findAll {it} parameters.get('labels',[]) + ['nodejs'] + ['ubuntu']
+    labels += [params.get('label')].findAll {it}  
```

Comment:
Would it not be better to create the labels variable with the default labels already set and just inject any additional ones based on the parameters? 

```
def labels = ['ubuntu', 'nodejs'] as Set
labels += params.get('label')
labels += parameters.get('labels',[])
labels - null
```

Whatever the logic is, you might want to consider moving it into a helper method.